portfolio/src/main/webapp/script.js

```diff
@@ -156,33 +156,63 @@ async function checkLogin(){
   }
 }
 
-/* Creates a map and adds it to the page. */
-function initMap() {
-  const jakarta = {lat: -6.175540, lng: 106.82743};
-  const zoom_scale = 5;
+function addUserLocation(){
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(async function(position) {
+      const params = new URLSearchParams();
+      params.append('lat', position.coords.latitude);
+      params.append('lng', position.coords.longitude);
+      fetch('/user-location-data', {method: 'POST', body: params});
+    }, function() {
```

Comment:
I am not sure but I think you are supposed to get some kind of cause of the error as a parameter here. Could you please check? If that's the case, it would be helpful to log the error as well maybe?